motionplan/motionPlanner.go

```diff
@@ -54,10 +54,46 @@ type PlanRequest struct {
 	Options            map[string]interface{}
 }
 
+// validatePlanRequest ensures PlanRequests are not malformed.
+func (req *PlanRequest) validatePlanRequest() error {
+	if req == nil {
+		return errors.New("PlanRequest cannot be nil")
+	}
+	if req.Logger == nil {
+		return errors.New("PlanRequest cannot have nil logger")
+	}
+
+	if req.Frame == nil {
+		return errors.New("PlanRequest cannot have nil frame")
+	}
+
+	if req.FrameSystem == nil {
+		return errors.New("PlanRequest cannot have nil framesystem")
+	} else if req.FrameSystem.Frame(req.Frame.Name()) == nil {
+		return frame.NewFrameMissingError(req.Frame.Name())
+	}
+
+	if req.Goal == nil {
+		return errors.New("PlanRequest cannot have nil goal")
+	} else if req.Goal != nil {
+		goalParentFrame := req.Goal.Parent()
+		if req.FrameSystem.Frame(goalParentFrame) == nil {
+			return frame.NewParentFrameMissingError(req.Goal.Name(), goalParentFrame)
+		}
+	}
+
+	if len(req.Frame.DoF()) != len(req.StartConfiguration[req.Frame.Name()]) {
```

Comment:
have we proved that are we sure that the `req.Frame.Name()` is in the `req.StartConfiguration` before we get to this point? Does this function still do the right thing if `req.Frame.Name()` is not in `req.StartConfiguration`?

Reply:
good catch, will add.

Reply:
Note that this existence check should only be applied in the case where `len(req.Frame.DoF()) > 0`. If the frame has 0 dof, then it will not (and should not) be in `StartConfiguration`